index.bs

```diff
@@ -371,28 +405,167 @@ model]]:
+    1. Set |navigable|'s [=top-level traversable/bounce tracking record=] to a new
+        [=bounce tracking record=] with [=bounce tracking record/initial host=]
+        set to |initialHost|.
+1. Otherwise,
+    1. If |sourceSnapshotParams|'s <a spec="html">has transient activation</a> is true:
+        1. Run [=record stateful bounces for bounce tracking=].
+        1. Set |navigable|'s [=top-level traversable/bounce tracking record=] to a new [=bounce tracking record=] with
+            [=bounce tracking record/initial host=] set to |initialHost|.
+    1. Otherwise, add |initialHost| to |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/bounce set=].
+1. Set |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/client bounce detection running=] to false.
+
+</div>
+
+<h5 id="bounce-tracking-mitigations-response-received-monkey-patch">Response Received Monkey
+Patch</h5>
+
+When the response is received at the end of a navigation, fill the [=bounce tracking record/bounce set=].
+
+Insert the following steps in the <a spec="html">navigate</a> algorithm, under "Create navigation params by fetching", after Step 19.7,
+"Wait until either response is non-null...", as soon as [=response/url list=] becomes available.
+
+1. Run [=process response received for bounce tracking=] given
+    <var ignore>navigable</var> and <var ignore>response</var>'s <var ignore>URL list</var>.
+
+<div algorithm>
+
+To <dfn>process response received for bounce tracking</dfn> given a [=navigable=]
+|navigable| and a [=list=] of [=URLs=] |URLs|, perform the following steps:
+
+1. If |navigable| is not a [=top-level traversable=], then abort these steps.
+1. [=Assert=]: |navigable|'s [=top-level traversable/bounce tracking record=] is not null.
+1. Run [=run steps after a timeout=] given:
+    1. <var ignore>global</var> = |navigable|'s [=navigable/active document=]'s [=relevant global object=]
+    1. <var ignore>milliseconds</var> = [=client bounce detection timer period=]
+    1. <var ignore>completionSteps</var> = [=record stateful bounces for bounce tracking=] given |navigable|.
+1. Set |navigable|'s [=top-level traversable/bounce tracking record=]'s
+    [=bounce tracking record/client bounce detection running=] to true.
+1. [=map/For each=] |URL| in |URLs|:
+    1. Let |site| be the result of running [=obtain a site=] given |URL|.
+    1. Let |host| be the |site|'s [=host=].
+    1. [=Insert=] |host| to the |navigable|'s [=top-level traversable/bounce tracking record=]'s [=bounce tracking record/bounce set=].
+
+</div>
+
+<h5 id="bounce-tracking-mitigations-document-load-monkey-patch">Document Loaded Monkey
+Patch</h5>
+
+When the document is loaded at the end of a navigation, update the [=bounce tracking record/final host=].
+
+Insert the following steps at the end of the <a spec="html">navigate</a> algorithm.
```

Comment:
Navigate has a couple ends. Is this
* as step 17, on the main thread
* as 16.11, in parallel
* as step 11 of https://html.spec.whatwg.org/multipage/browsing-the-web.html#finalize-a-cross-document-navigation, and similar last steps of the other "ending navigation" algorithms?